src/Admin/AbstractAdmin.php

```diff
@@ -3262,24 +3266,43 @@ public function canAccessObject($action, $object)
     }
 
     /**
-     * @return string[]
+     * @return MutableTemplateRegistryInterface
      */
-    protected function configureExportFields(): array
+    final public function getTemplateRegistry()
     {
-        return $this->getModelManager()->getExportFields($this->getClass());
+        // NEXT_MAJOR: Remove the deprecation and uncomment the exception.
+        if (!$this->hasTemplateRegistry()) {
+            @trigger_error(sprintf(
+                'Calling %s() when there is no template registry is deprecated since sonata-project/admin-bundle 3.x'
+                .' and will throw an exception in 4.0.'
+                .' Use %s::hasTemplateRegistry() to know if the template registry is set.',
+                __METHOD__,
+                __CLASS__
+            ), E_USER_DEPRECATED);
+        }
+        //if (false === $this->hasTemplateRegistry()) {
+        //    throw new \LogicException(sprintf('Unable to find the template registry for admin `%s`.', static::class));
+        //}
+
+        return $this->templateRegistry;
     }
 
-    protected function configureQuery(ProxyQueryInterface $query): ProxyQueryInterface
+    final public function hasTemplateRegistry(): bool
     {
-        return $query;
+        return $this->templateRegistry instanceof MutableTemplateRegistryInterface;
```

Comment:
Is there any constraint requiring `setTemplate()` or `setTemplates()` to be defined? If not, I think we should use `TemplateRegistryInterface` instead of `MutableTemplateRegistryInterface`.

Reply:
In 3.x branch:
```php
    public function setTemplates(array $templates)
    {
        // NEXT_MAJOR: Remove this line
        $this->templates = $templates;

        $this->getTemplateRegistry()->setTemplates($templates);
    }
```

Reply:
I understand that's the case for `3.x`, but then we should add a `NEXT_MAJOR` comment explaining how to proceed in the next major release.
The changes at `AdminInterface` should refer to `TemplateRegistryInterface`, right?

Reply:
$admin->getTemplateRegistry() should be equal to `admin_code.template_registry` service. It is created on:
```php 
final class TemplateRegistry implements final class TemplateRegistry implements MutableTemplateRegistryInterface
```
So `MutableTemplateRegistryInterface` will be correct.

Reply:
IMO we should check against null.

Reply:
Checking will be add in master. We can't throw exeption becouse someone compare `getTemplateRegistry()` to null. What we can do in 3.x is only add custom description for $this->getTemplateRegistry()->setTemplates($templates); exception.

Reply:
I mean: `return $this->... !== null`

Reply:
done

Reply:
I think @phansys is right, there is no need to return `MutableTemplateRegistryInterface` if we are not using `setTemplates` or `setTemplate`.

Reply:
@franmomu https://github.com/sonata-project/SonataAdminBundle/pull/6387#discussion_r488230331